A first cut at reinstating sharing for the documentation

rstento opened the pull request:
@RayRoestenburg & @andreaTP 

This is my first cut at making the files shareable with the commercial doc. There were many issues, one of which is that a page exists that is no longer referenced in the TOC, therefore the user has no way to get to it. It is ../administration/pages/more-storage-options.adoc. You'll find the content in the shared-content-source/modules folder. Please confirm whether this page is correct and should be included or not, in which case I'll delete it. 

Second, the docs were set up to use ROOT/partials/include.adoc to contain constants for all of the version numbers. There are two of these, one in docs and one in shared-content-source, which should be identical. I put the versions from versions.adoc into them, but I'm sure there are other places in the doc where the versions are hard-coded into pages. 

Third, I couldn't find any page that explains what cloudflow uses Kafka for, if one exists, could you please point me to it? 

Fourth, I believe the sections on Console and GCP should be removed? However, is the info in these sections more recent than what is in the enterprise doc? And, if the page on GCP cross-references this OSS doc instead of the commercial, that needs to be fixed before removing it.

I also would like to add the info that you gave me in the email (about installing components like Spark and Flink only if you need them and being able to use an existing Kafka) but I didn't have time to really think out where it should go and this PR needs to be merged before I can go into the enterprise doc and replace its out of date content with the pages on prerequisites and installing. So, I decided not to worry about it until I get that done.

rstento commented:
@RayRoestenburg, @andreaTP, The PR I've created in the commercial doc repo is dependent on this PR. In this PR, I've fixed a bunch of typos, xref issues etc. If you could review and merge when ready, that would be great. Then, we can move on to the other PR. Thanks!

andreaTP commented:
Going to review this tomorrow, thanks @rstento !

RayRoestenburg commented:
Thanks @rstento!

On Wed, 18 Nov 2020 at 19:44, Andrea Peruffo <notifications@github.com>
wrote:

> Going to review this tomorrow, thanks @rstento
> <https://github.com/rstento> !
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/lightbend/cloudflow/pull/863#issuecomment-729879547>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABGGCLA7QK77RL2S2TLPULSQQIZBANCNFSM4TTU4SYQ>
> .
>
-- 
*Cloudflow Tech Lead, Lightbend, Inc.*
ray@lightbend.com <raymond@lightbend.com>
@RayRoestenburg <https://twitter.com/RayRoestenburg>

<https://www.lightbend.com/>

andreaTP commented on docs/docs-source/docs/modules/ROOT/nav.adoc:
```diff
@@ -70,7 +69,8 @@ include::partial$include.adoc[]
-* xref:project-info:index.adoc[Project Information]
+* xref:project-info:index.adoc[Release notes and upgrading]
 ** xref:project-info:migration-1_3-2_0.adoc[Migration Guide 1.3.x to 2.0.x]
+** xref:administration:upgrading-cloudflow.adoc[Upgrading Cloudflow]
```
nice 👍

andreaTP commented on docs/docs-source/docs/modules/ROOT/partials/include.adoc:
```diff
@@ -1,5 +1,9 @@
-// global attributes
+
+// global attributes, if you update, make sure to update in shared-content-source ROOT:partials:include.adoc too
+:supported-java-v: >= 8
```
Global attributes can stay under this section:
https://github.com/lightbend/cloudflow/blob/master/docs/docs-source/site.yml#L23
and they will be propagated to all the docs automatically (at best of my understanding) without the need of a comment.

rstento replied:
I've found that authors don't tend to review the site.yml before they do a pull because it has a lot of other stuff in it. I've had better results with a separate include file. And, for some reason, I've had a problem with Antora propagating them correctly from site.yml. However, if @RayRoestenburg and @andreaTP prefer the site file. I'll test it today and make sure it works.

rstento replied:
I tried to move the global attributes into site.yml. However, because of the way I'm sharing the content, Antora seems to generate HTML for shared content without looking at them. So, the build is filled with errors. There is another way to share content, but it would take a lot of work and we don't know whether we will be sharing later down the road. So, I've left them in the include files for now.

andreaTP commented on docs/docs-source/docs/modules/console/pages/index.adoc:
```diff
@@ -5,11 +5,8 @@
 include::ROOT:partial$include.adoc[]
 
-Available with a Lightbend Platform subscription, Lightbend Console provides deep insights and observability for Cloudflow applications. It helps you verify that your application looks as you expect--with the correct streamlets and the proper connections, each using the desired schema. In the Console, you can watch your application run, check on key performance indicators, such as latency and thr ...
+Available with a Lightbend Platform subscription and on Google Cloud Platform marketplace, Lightbend Console provides deep insights and observability for Cloudflow applications. It helps you verify that your application looks as you expect--with the correct streamlets and the proper connections, each using the desired schema. In the Console, you can watch your application run, check on key perfor ...
```
cc. @RayRoestenburg

RayRoestenburg replied:
We are not ready for Google Cloud Platform marketplace yet, so it is better to keep it out for now.

rstento replied:
Oh, news to me. Thanks! I'll comment it out or remove where necessary. I guess that means the whole section on GCP needs to go too.

andreaTP commented on docs/shared-content-source/docs/modules/ROOT/pages/app-development-process.adoc:
```diff
@@ -250,8 +248,9 @@ Next, we see how to use the application descriptor to deploy the application to
-In contrast with the usual YAML-full experience that typical K8s deployments require, Cloudflow uses the blueprint information and the streamlet definitions to auto-generate an application deployment. 
+After testing and packaging, you are ready to deploy the application to a Cloudflow-enabled Kubernetes cluster. A Cloudflow-enabled Kubernetes cluster has Cloudflow installed as well as Spark and/or Flink, if your streamlets use them. In contrast with the usual YAML-full experience that typical K8s deployments require, Cloudflow uses the blueprint information and the streamlet definitions to auto ...
+
+Kafka is used by Cloudflow to connect streamlets together in a blueprint. If you intend to connect streamlets in this way, at least one Kafka cluster should be available before you install Cloudflow. Cloudflow may be used without Kafka (for example, when your application contains a single streamlet, or an xref:develop:clustering-akka-streamlet.adoc[Akka cluster]), but if your team intends to conn ...
```
> together in a blueprint
something like:
"with the topology described in the blueprint"

> If you intend to connect streamlets in this way
well, there is no supported "other way" ATM

RayRoestenburg replied:
Important to note that before, the blueprint would connect streamlet with streamlet through inlets and outlets through a connection section. Since 2.0 streamlet inlets and outlets are consumers and producers respectively of topics. And, BTW, a streamlet does not have to have inlets or outlets. It could use some other message queue or database to read from and write to.

rstento replied:
I added "with the topoloty described in the blueprint" but I'm still not sure how to modify the rest of the para.

andreaTP commented on docs/shared-content-source/docs/modules/ROOT/partials/include.adoc:
```diff
@@ -1,4 +1,34 @@
+:supported-helm-v: version 3 or higher
+
+:kubernetes-platform-name:     Kubernetes
+:kubernetes-platform-version:  1.12
```
this looks pretty outdated actually, I would omit the Kubernetes version since it will require continuous maintenance

RayRoestenburg replied:
yes, agree.

rstento replied:
So, I removed kubernetes-platform-version.

andreaTP commented on docs/shared-content-source/docs/modules/ROOT/partials/include.adoc:
```diff
@@ -1,4 +1,34 @@
+:minimum-openshift-platform-version:       3.11
+:minimum-platform-version-major-dot-minor: 3.11
+:minimum-sbt-version:                      0.13.7
+:latest-sbt-version: 					   1.3.8	
```
The latest sbt 1.3.X is 1.3.13 but we are likely to support 1.4.X anytime soon after this get merged:
https://github.com/lightbend/cloudflow/pull/880

rstento replied:
I'll update to 1.3.13, pls let me know when we support 1.4.X

andreaTP replied:
since, well, next version :-)

andreaTP commented on docs/shared-content-source/docs/modules/administration/pages/how-to-install-and-use-strimzi.adoc:
```diff
@@ -0,0 +1,136 @@
+
+We are going to create Strimzi in the `cloudflow` namespace in this guide. Make sure that the namespace `cloudflow` exists before continuing. To create the `cloudflow` namespace, execute
+
+  kubectl create ns cloudflow
```
is there a way for having those commands coming from a Makefile?
i.e.:
https://github.com/lightbend/cloudflow/blob/master/integration-tests/Makefile#L36

rstento replied:
If it is in the examples/doc-snippets dir. I'll check and see.

rstento replied:
This file is not in the examples/doc-snippets dir, but could be added there.

andreaTP commented on docs/shared-content-source/docs/modules/administration/pages/how-to-install-and-use-strimzi.adoc:
```diff
@@ -0,0 +1,136 @@
+You can use the following command to create a cluster:
+
+[source,shell script]
+----
```
also the files ideally should come from:
https://github.com/lightbend/cloudflow/blob/master/integration-tests/kafka-cluster.yaml

RayRoestenburg replied:
I could imagine that the integration tests would do slightly different things. Also, I think with antora the snippets need to be in a specific place, sadly

rstento replied:
yes, examples need to be in examples/doc-snippets. However, a soon-coming release of antora should remove this limitation.

andreaTP replied:
please keep us posted when the limitation is removed!

andreaTP commented on docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration.adoc:
```diff
@@ -83,7 +83,10 @@ cloudflow {
-Environment / shell variables can be specified in the config as `${variable_name}`, `$variable_name`, or `${?variable_name}`. 
+Environment / shell variables can be specified in the config as `$\{variable_name}`, `$variable_name`, or `${?variable_name}`. 
+
+ifdef::review[REVIEWERS: I needed to escape the {} in the first variable_name above to prevent antora from complaining about an undefined variable.]
```
perfectly aligned:
https://asciidoctor.org/docs/user-manual/#preventing-substitutions
but I think it should be escaped `${?variable_name}` as well, or am I wrong?

rstento replied:
It may be a bug in Antora, but I only needed to escape the first one. When i tried to do the remainder, the / appeared in the HTML output.

rstento replied:
Tried it again and still only need to escape the first.

andreaTP replied:
I trust you Ruth :-)

andreaTP commented on docs/shared-content-source/docs/modules/get-started/pages/prepare-development-environment.adoc:
```diff
@@ -1,22 +1,24 @@
+* Java(JDK) {supported-java-v}
+* https://www.scala-sbt.org/[sbt,window=_blank], {supported-sbt-v}
+* https://www.docker.com/community-edition[Docker,window=_blank], {supported-docker-v}
+* https://helm.sh/docs/intro/install/[Helm,window=_blank], {supported-helm-v}. We recommend version 3 to avoid dealing with tiller.
```
I think it's ok to drop Helm v2 support, WDYT @RayRoestenburg ?

RayRoestenburg replied:
yes, drop it.

RayRoestenburg replied:
It will just cause a lot of pain if we do say we support 2.

rstento replied:
Will do.

RayRoestenburg commented on docs/docs-source/docs/modules/ROOT/pages/index.adoc:
```diff
@@ -62,8 +62,9 @@ Cloudflow includes tools for developing and deploying streaming applications:
-Lightbend also offers support with a https://www.lightbend.com/lightbend-platform-subscription[Lightbend Subscription].
+Lightbend's commercial version of Cloudflow, https://developer.lightbend.com/docs/cloudflow/current/index.html[Akka Data Pipelines], includes support along with https://developer.lightbend.com/docs/telemetry/current/home.html[Telemetry] and https://developer.lightbend.com/docs/cloudflow/current/console/index.html[Lightbend Console], which provide visibility into your running Cloudflow apps.
+
+You can also use https://developer.lightbend.com/docs/cloudflow/current/gcp/index.html[Akka Data Pipelines on Google Cloud Platform]. 
```
I would remove this for now it is not ready.

RayRoestenburg replied:
The version on GCP Marketplace is outdated and can only be installed on an existing cluster.

rstento replied:
OK, I'll comment out for now.

RayRoestenburg commented on docs/docs-source/docs/modules/administration/pages/index.adoc:
```diff
@@ -1,23 +1,22 @@
 include::ROOT:partial$include.adoc[]
 
-This chapter describes common Cloudflow and Kubernetes administration tasks.
+When you develop your Cloudflow application, you have the option to use Akka Streams, Spark, and Flink. Cloudflow relies on Kafka for persisting messages. This chapter contains the steps to install Cloudflow, using the `Helm` and `kubectl` command-line tools. It also shows how to install Kafka, Spark, and Flink operators that integrate with Cloudflow and how to expose your services to external ac ...
```
```suggestion
When you develop your Cloudflow application, you have the option to use Akka Streams, Spark, Flink and Kafka. This chapter contains the steps to install Cloudflow, using the `Helm` and `kubectl` command-line tools. It also shows how to install Kafka, Spark, and Flink operators that integrate with Cloudflow and how to expose your services to external access.
```

rstento replied:
Will do.

RayRoestenburg commented on docs/docs-source/docs/modules/console/pages/index.adoc:
```diff
@@ -5,11 +5,8 @@
 include::ROOT:partial$include.adoc[]
 
-Available with a Lightbend Platform subscription, Lightbend Console provides deep insights and observability for Cloudflow applications. It helps you verify that your application looks as you expect--with the correct streamlets and the proper connections, each using the desired schema. In the Console, you can watch your application run, check on key performance indicators, such as latency and thr ...
+Available with a Lightbend Platform subscription and on Google Cloud Platform marketplace, Lightbend Console provides deep insights and observability for Cloudflow applications. It helps you verify that your application looks as you expect--with the correct streamlets and the proper connections, each using the desired schema. In the Console, you can watch your application run, check on key perfor ...
```
```suggestion
Available with a Lightbend Platform subscription, Lightbend Console provides deep insights and observability for Cloudflow applications. It helps you verify that your application looks as you expect--with the correct streamlets and the proper connections, each using the desired schema. In the Console, you can watch your application run, check on key performance indicators, such as latency and throughput, and observe the health of each streamlet and the monitors computed on them.
```

rstento replied:
Will do.

rstento replied:
I removed the whole console section, since it is in the Enterprise doc and is a commercial feature.

RayRoestenburg commented on docs/docs-source/docs/modules/get-started/pages/index.adoc:
```diff
@@ -6,26 +6,29 @@
+.. xref:administration:index.adoc[Installing Cloudflow]
+.. xref:deploy-to-k8s-cluster.adoc[Deploy to a Kubernetes cluster]
+
+ifdef::review[REVIEWERS: the following are fine-grained tasks, not top-level tasks and since they are covered in the pages above and don't have their own page, I think it's best to leave them off this list.]
```
Sounds good.

rstento commented:
Thanks for the review @andreaTP and @RayRoestenburg! I should submit another commit today with the changes.

rstento commented:
@andreaTP @RayRoestenburg, I hope I addressed all of your comments correctly, The build is still failing and I don't know why. I can build it just fine locally.

andreaTP commented on docs/docs-source/docs/modules/ROOT/pages/index.adoc:
```diff
@@ -50,7 +50,7 @@ image::apps-1.png[]
 == Feature set
 
-Cloudflow includes tools for developing and deploying streaming applications:
+Cloudflow dramatically accelerates your application development efforts, reducing the time required to create, package, and deploy an application from weeks--to hours. Cloudflow includes tools for developing and deploying streaming applications:
```
`development` / `developing` / `development` in 3 following sentences, it's not introduced in this PR, but would be nice to avoid the repetition.

andreaTP commented on docs/shared-content-source/docs/modules/ROOT/partials/include.adoc:
```diff
@@ -1,4 +1,35 @@
+:minimum-kubernetes-on-openshift-version:  1.11
+:minimum-openshift-platform-version:       3.11
+:minimum-platform-version-major-dot-minor: 3.11
+:minimum-sbt-version:                      0.13.7
```
not introduced in this PR, but this is outdated ... cc. @RayRoestenburg

andreaTP commented on docs/shared-content-source/docs/modules/ROOT/partials/include.adoc:
```diff
@@ -1,4 +1,35 @@
+:latest-sbt-version: 					   1.3.13
+
+
+:supported-akka-v: 2.6.6 
```
not introduced in this PR, but this is outdated ... cc. @RayRoestenburg 
we need to find a "sane" way of keeping those numbers updated ...

andreaTP reviewed (approved):
A couple of minor comments but looks good to me.
@RayRoestenburg if you are good with these changes I can take over this PR, check, fix what is breaking, and eventually rebase on top of the latest master branch to get it merged.
Is it ok for you @rstento ?

rstento commented:
> A couple of minor comments but looks good to me.
> @RayRoestenburg if you are good with these changes I can take over this PR, check, fix what is breaking, and eventually rebase on top of the latest master branch to get it merged.
> Is it ok for you @rstento ?

Fine with me @andreaTP, Thanks!

RayRoestenburg commented:
Yes sounds good @andreaTP

andreaTP commented:
merged in #901 , thanks a lot @rstento !
